### PR TITLE
[framework] complaint mail templates

### PR DESCRIPTION
**Pull request by AndrejBlaho**
#### Description, the reason for the PR
with addition of complaints there was need for new mail templates handling communication regarding changes in complaint status

#### Fixes issues
... 

#### Check the appropriate checkboxes below, please

- [x] [BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/) 
- [ ] New feature 
- [x] have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)

## Summary by CodeRabbit

- **New Features**
	- Enhanced email notification system for complaint status changes.
	- Added support for complaint status in mail templates, improving email communication related to complaints.
	- Introduced new columns and properties to facilitate better management of complaint-related emails.
	- Added translations for complaint-related terms in Czech and English.

- **Bug Fixes**
	- Improved logic for disabling email sending options in the mail template configuration.

- **Documentation**
	- Updated documentation to reflect new complaint handling features.

- **Chores**
	- Updated service definitions and localization files to support new complaint handling features.

----
:globe_with_meridians: Live Preview:
  - https://ab-complaint-email-templates.odin.shopsys.cloud
  - https://cz.ab-complaint-email-templates.odin.shopsys.cloud

### CODE REVIEW

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+     * @param \Shopsys\FrameworkBundle\Component\Router\DomainRouterFactory $domainRouterFactory
+     * @param \Twig\Environment $twig
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Shopsys\FrameworkBundle\Twig\PriceExtension $priceExtension
+     * @param \Shopsys\FrameworkBundle\Twig\DateTimeFormatterExtension $dateTimeFormatterExtension
+     */
+    public function __construct(
+        protected readonly Setting $setting,
+        protected readonly DomainRouterFactory $domainRouterFactory,
+        protected readonly Environment $twig,
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Complaint/Mail/ComplaintMail.php]: this property seems unused

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        protected readonly Domain $domain,
+        protected readonly PriceExtension $priceExtension,
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Complaint/Mail/ComplaintMail.php]: this property seems unused

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Mail\MailTemplate $mailTemplate
+     * @param \Shopsys\FrameworkBundle\Model\Complaint\Complaint $complaint
+     * @return \Shopsys\FrameworkBundle\Model\Mail\MessageData
+     */
+    public function createMessage(MailTemplate $mailTemplate, $complaint)
+    {
+        $orderDomainId = $complaint->getOrder()->getDomainId();
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Complaint/Mail/ComplaintMail.php]: shouldn't it be `$complaintId = $complaint->getId()`?

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Complaint\Complaint $complaint
+     * @return array
+     */
+    protected function getVariablesReplacementsForBody(Complaint $complaint)
+    {
+        $order = $complaint->getOrder();
+        $orderDomainId = $order->getDomainId();
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Complaint/Mail/ComplaintMail.php]: shouldn't it be complaintDomainId?

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+        $router = $this->domainRouterFactory->getRouter($orderDomainId);
+
+        return [
+            self::VARIABLE_COMPLAINT_NUMBER => htmlspecialchars($complaint->getNumber(), ENT_QUOTES),
+            self::VARIABLE_ORDER_NUMBER => htmlspecialchars($complaint->getOrder()->getNumber(), ENT_QUOTES),
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Complaint/Mail/ComplaintMail.php:83]: there is already $order variable defined earlier

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+            self::VARIABLE_DATE => $this->getFormattedDateTime($complaint),
+            self::VARIABLE_URL => $router->generate('front_homepage', [], UrlGeneratorInterface::ABSOLUTE_URL),
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Complaint/Mail/ComplaintMail.php:85]: shouldn't be the URL somewhere different than the front page?

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+        );
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Complaint\Complaint $complaint
+     * @return string
+     */
+    protected function getDomainLocaleByComplaint(Complaint $complaint)
+    {
+        return $this->domain->getDomainConfigById($complaint->getOrder()->getDomainId())->getLocale();
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Complaint/Mail/ComplaintMail.php]: complaint domain id?

File: packages/framework/src/Model/Complaint/Mail/ComplaintMailFacade.php [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        protected readonly MailTemplateFacade $mailTemplateFacade,
+        protected readonly ComplaintMail $complaintMail,
+        protected readonly UploadedFileFacade $uploadedFileFacade,
+    ) {
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Complaint\Complaint $complaint
+     */
+    public function sendEmail(Complaint $complaint): void
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Complaint/Mail/ComplaintMailFacade.php:33]: shouldn't it be used somewhere?

**Review by grossmannmartin — DISMISSED** [review]

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php (hunk 1 above)
Comment (AndrejBlaho) [reply packages/framework/src/Model/Complaint/Mail/ComplaintMail.php]: removed

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php (hunk 2 above)
Comment (AndrejBlaho) [reply packages/framework/src/Model/Complaint/Mail/ComplaintMail.php]: removed

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php (hunk 3 above)
Comment (AndrejBlaho) [reply packages/framework/src/Model/Complaint/Mail/ComplaintMail.php]: yes, missed this while rebasing, before there was no complaint domain if im not mistaken

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php (hunk 4 above)
Comment (AndrejBlaho) [reply packages/framework/src/Model/Complaint/Mail/ComplaintMail.php]: yes, same as above

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php (hunk 5 above)
Comment (AndrejBlaho) [reply packages/framework/src/Model/Complaint/Mail/ComplaintMail.php:83]: is it? im only seeing it @ getVariablesReplacementsForSubject method, but i thought those are separate things

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php (hunk 6 above)
Comment (AndrejBlaho) [reply packages/framework/src/Model/Complaint/Mail/ComplaintMail.php:85]: im not sure, i based this after order, what would you recommend? URL for complaint itself?

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php (hunk 7 above)
Comment (AndrejBlaho) [reply packages/framework/src/Model/Complaint/Mail/ComplaintMail.php]: yes 😅

File: packages/framework/src/Model/Complaint/Mail/ComplaintMailFacade.php (hunk 8 above)
Comment (AndrejBlaho) [reply packages/framework/src/Model/Complaint/Mail/ComplaintMailFacade.php:33]: you are right 😅 added, should it be somewhere else as well?

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php (hunk 6 above)
Comment (vitek-rostislav) [reply packages/framework/src/Model/Complaint/Mail/ComplaintMail.php:85]: yeah, this URL is ok, however, please add `{complaint_detail_url}` variable as well :pray:

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]
please add `{complaint_detail_url}` variable 

// Edit: will be done in another user story

File: packages/framework/src/Model/Complaint/Mail/ComplaintMail.php (hunk 6 above)
Comment (vitek-rostislav) [reply packages/framework/src/Model/Complaint/Mail/ComplaintMail.php:85]: ah, it will be in another user story

**Review by vitek-rostislav — APPROVED** [review]

File: packages/framework/src/Model/Complaint/Status/ComplaintStatusFacade.php [hunk 9]
```diff
@@ -33,6 +36,10 @@ public function create(ComplaintStatusData $complaintStatusData): ComplaintStatu
         $this->em->persist($complaintStatus);
         $this->em->flush();
 
+        $this->mailTemplateFacade->createMailTemplateForAllDomains(
```
Comment (vitek-rostislav) [inline packages/framework/src/Model/Complaint/Status/ComplaintStatusFacade.php:40]: the complaint status is not associated with the mail template this way Bw, the same problem is with the order statuses. I suggest fixing both usages in a separate commit by modifying the method like this:
```
public function createMailTemplateForAllDomains(string $name, ?OrderStatus $orderStatus = null, ?ComplaintStatus $complaintStatus = null): void
```

do not forget the upgrade notes then, please :pray:

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]

File: packages/framework/src/Model/Complaint/Status/ComplaintStatusFacade.php (hunk 9 above)
Comment (AndrejBlaho) [reply packages/framework/src/Model/Complaint/Status/ComplaintStatusFacade.php:40]: I have changed it, is it ok this way?

Comment (vitek-rostislav) [reply packages/framework/src/Model/Complaint/Status/ComplaintStatusFacade.php:40]: yes, awesome!

**Review by vitek-rostislav — APPROVED** [review]

**Pull request merged**